staging/src/k8s.io/client-go/examples/leader-election/main.go

```diff
@@ -124,38 +121,29 @@ func main() {
 		// get elected before your background loop finished, violating
 		// the stated goal of the lease.
 		ReleaseOnCancel: true,
-		LeaseDuration:   60 * time.Second,
-		RenewDeadline:   15 * time.Second,
-		RetryPeriod:     5 * time.Second,
+		LeaseDuration:   15 * time.Second,
+		RenewDeadline:   10 * time.Second,
+		RetryPeriod:     2 * time.Second,
 		Callbacks: leaderelection.LeaderCallbacks{
 			OnStartedLeading: func(ctx context.Context) {
 				// we're notified when we start - this is where you would
 				// usually put your code
-				klog.Infof("%s: leading", id)
+				run(ctx)
 			},
 			OnStoppedLeading: func() {
-				// we can do cleanup here, or after the RunOrDie method
-				// returns
-				klog.Infof("%s: lost", id)
+				// we can do cleanup here
+				klog.Fatalf("leader lost: %s", id)
```

Comment:
why fatal here?

Reply:
only one place to put clean up code is better for reading.

Reply:
exiting when leader lease is lost is reasonable, but this exits with a non-zero return code, which we probably don't want